modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/translation/TranslationModal.js

```diff
@@ -0,0 +1,264 @@
+									return (
+										<ClayDropDown.Item
+											key={locale.label}
+											onClick={() => {
+												restoreLocale(locale);
+											}}
+											symbolLeft={locale.symbol}
+										>
+											{locale.label}
+										</ClayDropDown.Item>
+									);
+								})}
+							</ClayDropDown.ItemList>
+						</ClayDropDown>
+					</ClayManagementToolbar.ItemList>
+				</ClayManagementToolbar>
+
+				{deletedLocale && (
+					<ClayAlert
+						displayType="success"
+						onClose={() => {
+							setDeletedLocale(null);
+						}}
+						title={Liferay.Language.get('success')}
+					>
+						{Liferay.Util.sub(
+							Liferay.Language.get('translation-deleted'),
+							deletedLocale.label
+						)}
+						<ClayLink
+							onClick={() => {
+								restoreLocale(deletedLocale);
+							}}
+						>
+							{Liferay.Language.get('undo')}
+						</ClayLink>
+					</ClayAlert>
+				)}
+
+				<ClayTable>
+					<ClayTable.Head>
+						<ClayTable.Row>
+							<ClayTable.Cell headingCell>
+								{Liferay.Language.get('code')}
+							</ClayTable.Cell>
+							<ClayTable.Cell headingCell>
+								{Liferay.Language.get('language')}
+							</ClayTable.Cell>
+							<ClayTable.Cell headingCell>
+								{Liferay.Language.get('status')}
+							</ClayTable.Cell>
+							<ClayTable.Cell headingCell />
+						</ClayTable.Row>
+					</ClayTable.Head>
+					<ClayTable.Body>
+						{filteredLocales.map((locale, index) => {
+							const label = locale.label;
+
+							const isDefaultLocale =
+								locale.label === defaultLocale.label;
+							const value = translations[label];
+
+							return (
+								<ClayTable.Row key={label}>
+									<ClayTable.Cell>
+										<>
+											<ClayIcon
+												className="inline-item inline-item-before"
+												symbol={locale.symbol}
+											/>
+											<strong>{label}</strong>
+										</>
+									</ClayTable.Cell>
+									<ClayTable.Cell expanded>
+										{locale.displayName}
+									</ClayTable.Cell>
+									<ClayTable.Cell>
+										<ClayLabel
+											displayType={
+												isDefaultLocale
+													? 'info'
+													: value
+													? 'success'
+													: 'warning'
+											}
+										>
+											{isDefaultLocale
+												? ariaLabels.default
+												: value
+												? ariaLabels.translated
+												: ariaLabels.untranslated}
+										</ClayLabel>
+									</ClayTable.Cell>
+									<ClayTable.Cell>
+										{!isDefaultLocale && (
+											<ClayIcon
+												className="inline-item"
+												onClick={() => {
+													filteredLocales.splice(
+														index,
+														1
+													);
+
+													deletedLocales.add(locale);
+													setDeletedLocales(
+														deletedLocales
+													);
+
+													setDeletedLocale(locale);
+
+													setInitialLocales(
+														filteredLocales
+													);
```

Comment:
I think this logic could be simplified a bit, for example `filteredLocales` doesn't need to be modified, because we just need to filter the `initialLocales` by removing the locale and `filteredLocales` will be updated on rendering.

```js
deletedLocales.add(locale);
setDeletedLocales(
  deletedLocales
);

setDeletedLocale(locale);

setInitialLocales(
  initialLocales.filter(({label}) => label !== locale.label)
);
```

It just avoids the side effect by directly changing the value of `filteredLocales` and it also avoids losing locales when the user is searching something and we use it to update the `initialLocales`.